airbyte-integrations/connectors/destination-gcs-data-lake/src/main/kotlin/io/airbyte/integrations/destination/gcs_data_lake/catalog/GcsDataLakeTableIdGeneratorFactory.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.gcs_data_lake.catalog
+
+import io.airbyte.cdk.load.command.DestinationStream
+import io.airbyte.cdk.load.data.Transformations
+import io.airbyte.cdk.load.orchestration.db.FinalTableNameGenerator
+import io.airbyte.cdk.load.table.TableName
+import io.airbyte.cdk.load.toolkits.iceberg.parquet.TableIdGenerator
+import io.airbyte.cdk.load.toolkits.iceberg.parquet.tableIdOf
+import io.airbyte.integrations.destination.gcs_data_lake.spec.GcsDataLakeConfiguration
+import io.micronaut.context.annotation.Factory
+import javax.inject.Singleton
+import org.apache.iceberg.catalog.TableIdentifier
+
+/**
+ * Sanitizes a name to be BigLake/BigQuery compatible.
+ * - Converts to alphanumeric + underscore only
+ * - Prefixes with underscore if it starts with a number
+ */
+private fun sanitizeBigLakeName(name: String): String {
```

Comment:
nit: rename to sanitizeBigLakeTableName? Or put this in a file that isn't titled "table ID generator", if you want to also use it in the column name generator